launch/shell.go

```diff
@@ -92,11 +57,31 @@ func (l *Launcher) profiles(process Process) ([]string, error) {
 	} else if err != nil {
 		return nil, errors.Wrapf(err, "failed to determine if app profile script exists at path '%s'", filepath.Join(l.AppDir, appProfile))
 	}
-	appendIfFile(filepath.Join(l.AppDir, appProfile), fi)
+	if !fi.IsDir() {
+		profiles = append(profiles, appProfile)
+	}
 
 	return profiles, nil
 }
 
+func (l *Launcher) populateLayerProfiles(proc Process, profiles *[]string) action {
+	return func(path string) error {
+		if err := eachFile(filepath.Join(path, "profile.d"), func(path string) error {
+			*profiles = append(*profiles, path)
```

Comment:
```suggestion
		if err := eachFile(filepath.Join(path, "profile.d"), func(profilePath string) error {
			*profiles = append(*profiles, profilePath)
```

Reply:
I renamed the other use of path to `layerDir` which I think should make this use of `path` clearer.